providers/mongodb.js

```diff
@@ -85,3 +85,18 @@ module.exports = class extends Provider {
 };
 
 const resolveQuery = query => isObject(query) ? query : { id: query };
+
+const parseUpdateObject = (doc, pref = '', oldObj = {}) => {
+	const obj = oldObj;
+	const prefix = pref !== '' ? `${pref}.` : '';
+	for (const key in doc) {
+		if (Object.prototype.hasOwnProperty.call(doc, key)) {
+			if (typeof doc[key] !== 'object' || Object.keys(doc[key]).length === 0) {
```

Comment:
use `isObject`
`null` is typeof object.
`typeof null === "object";` will be true